File: anika.dovob/src/main/java/com/example/anika/example/entity/Order.java
package com.example.anika.example.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "ORDERS")
@Getter
@Setter
public class Order {

    @Id
    @GeneratedValue
    @Column(name = "ORDER_ID")
    private Long id;

    @Column(name = "MEMBER_ID")
    private Long memberId;

    /**
     * NOTE:
     *
     * @Temporal은 Java 8+ java.time API 도입으로 deprecated됨
     * LocalDate, LocalDateTime 등 타입 자체가 DATE/TIMESTAMP를 명시하므로 불필요
     */
    // @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "ORDER_DATE")
    private LocalDateTime orderDate;

    @Enumerated(EnumType.STRING)
    private OrderStatus status;
}

File: jayy/src/main/kotlin/com/study/jpa/chapter06/homework3/Order.kt
package com.study.jpa.chapter06.homework3

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity(name = "Hw3OrderKt")
@Table(name = "orders_hw3_kt")
class Order(
    @Id
    @GeneratedValue
    @Column(name = "order_id")
    val id: Long? = null,

    @ManyToOne
    @JoinColumn(name = "member_id")
    var member: Member? = null,  // 다대다를 중간 엔티티로 풀기: Member -> Order

    @ManyToOne
    @JoinColumn(name = "product_id")
    var product: Product? = null,  // 다대다를 중간 엔티티로 풀기: Product -> Order

    var count: Int = 0,              // 주문 수량

    var orderDate: LocalDateTime? = null  // 주문 일자
)
